tests/golem/envs/localhost.py

```diff
@@ -0,0 +1,303 @@
+import asyncio
+import logging
+from enum import Enum
+from pathlib import Path
+from threading import Thread
+from typing import Optional, Dict, Any, Tuple, List
+
+from dataclasses import dataclass, asdict, field
+from golem_task_api import RequestorAppHandler, ProviderAppHandler
+from golem_task_api.server import (
+    AppServer,
+    ProviderAppServer,
+    RequestorAppServer
+)
+from golem_task_api.structs import Subtask
+from twisted.internet import defer, threads
+
+from golem.envs import (
+    CounterId,
+    CounterUsage,
+    EnvConfig,
+    EnvId,
+    EnvironmentBase,
+    EnvMetadata,
+    EnvSupportStatus,
+    Prerequisites,
+    Runtime,
+    RuntimeBase,
+    RuntimeInput,
+    RuntimeOutput,
+    RuntimePayload
+)
+from golem.task.task_api import TaskApiPayloadBuilder
+
+logger = logging.getLogger(__name__)
+
+
+class LocalhostConfig(EnvConfig):
+
+    def to_dict(self) -> dict:
+        return {}
+
+    @staticmethod
+    def from_dict(data: dict) -> 'LocalhostConfig':
+        return LocalhostConfig()
+
+
+@dataclass
+class LocalhostPrerequisites(Prerequisites):
+    compute_results: Dict[str, str] = field(default_factory=dict)
+    benchmark_result: float = 0.0
+    subtasks: List[Subtask] = field(default_factory=list)
+    verify_results: Dict[str, Tuple[bool, Optional[str]]] \
+        = field(default_factory=dict)
+
+    def to_dict(self) -> dict:
+        return asdict(self)
+
+    @staticmethod
+    def from_dict(data: dict) -> 'LocalhostPrerequisites':
+        return LocalhostPrerequisites(**data)
+
+
+class ServiceType(Enum):
+    PROVIDER = 'provider'
+    REQUESTOR = 'requestor'
+
+
+@dataclass
+class LocalhostPayload(RuntimePayload):
+    service_type: ServiceType
+    port: int
+    shared_dir: Path
+    prerequisites: LocalhostPrerequisites
+
+
+class LocalhostPayloadBuilder(TaskApiPayloadBuilder):
+
+    @classmethod
+    def create_payload(
+            cls,
+            prereq: Prerequisites,
+            shared_dir: Path,
+            command: str,
+            port: int
+    ) -> RuntimePayload:
+        assert isinstance(prereq, LocalhostPrerequisites)
+        # Parse the command e.g. 'provider 1234' -> 'PROVIDER'
+        service_type, _ = command.upper().split()
```

Comment:
You can ignore a variable count of the arguments with
```suggestion
        service_type, *_ = command.upper().split()
```